### PR TITLE
feat: Support HelmAddon strategy to deploy NFD

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

File: test/e2e/nfd_helpers.go [hunk 1]
```diff
@@ -34,15 +36,35 @@ func WaitForNFDToBeReadyInWorkloadCluster(
...
+		waitForClusterResourceSetToApplyResourcesInCluster(
+			ctx,
+			waitForClusterResourceSetToApplyResourcesInClusterInput{
+				name:         "node-feature-discovery-" + input.WorkloadCluster.Name,
+				clusterProxy: input.ClusterProxy,
+				cluster:      input.WorkloadCluster,
+				intervals:    input.ClusterResourceSetIntervals,
+			},
+		)
+	case v1alpha1.AddonStrategyHelmAddon:
```
Comment (dkoshkin) [inline test/e2e/nfd_helpers.go:50]: I love how easy it is to extend the tests.

File: pkg/handlers/generic/lifecycle/nfd/strategy_crs.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cluster.Namespace,
+			Name:      defaultCM.Name,
+		},
+	}
+	cm.Data = maps.Clone(defaultCM.Data)
```
Comment (dkoshkin) [inline pkg/handlers/generic/lifecycle/nfd/strategy_crs.go:89]: Just curious whats the benefit of this over just setting
```
cm.Data = defaultCM.Data
```

File: pkg/handlers/generic/lifecycle/nfd/handler.go [hunk 3]
```diff
@@ -81,77 +74,51 @@ func (n *DefaultNFD) AfterControlPlaneInitialized(
...
+		}
+	case v1alpha1.AddonStrategyHelmAddon:
+		strategy = helmAddonStrategy{
+			config: n.config.helmAddonConfig,
+			client: n.client,
+		}
+	default:
 		resp.SetStatus(runtimehooksv1.ResponseStatusFailure)
-		log.Error(err, "failed to apply NFD ConfigMap for cluster")
+		resp.SetMessage(fmt.Sprintf("unknown NFD addon deployment strategy %q", cniVar.Strategy))
```
Comment (dkoshkin) [inline pkg/handlers/generic/lifecycle/nfd/handler.go:113]: Nice! We need to get better at returning back errors

File: pkg/handlers/generic/lifecycle/nfd/strategy_crs.go (hunk 2 above)
Comment (jimmidyson) [reply pkg/handlers/generic/lifecycle/nfd/strategy_crs.go:89]: Good question! None! I'll change it

Comment (jimmidyson) [reply pkg/handlers/generic/lifecycle/nfd/strategy_crs.go]: Done.

**Pull request merged**